### DISCUSSION TITLE
Is it even an option to use Blazor without Azure?

### DISCUSSION THREAD

**Discussion by iam3yal** [dotnet/aspnetcore#28506] [General] [answered]
Hello, 

I'm trying to figure how expensive is to run Blazor on the server comparing to other SPA frameworks and it seems like results push me into Azure (probably due to SignalR) which I don't want to do at least initially as it seems pretty expensive to start with so my setup/VPS for now looks something like the following: 

1. x1 to x2 CPUs at 5 GHz.
2. x2 to x6 gigs of memory.
3. Bandwidth of 5000 gigs per month over a 10 Gb/s connection.
4. Windows Server 19 and SQL Server 19 Standard.

The question is whether **can I even use a VPS (with IIS) and run Blazor as if it was just a regular Web App** or this is one of the pitfalls of using Blazor and Azure is a hidden dependency? I can't estimate how many concurrent connections I'd have as I'm just starting out but I guess it would be pretty small amount in the hundreds to thousands area and maybe I'm just too optimistic here.

It's really unfortunate that most of the docs that I've read tell you to use Azure for concurrency but then if you just want to start out and start small maybe you don't have the funds and maybe it's not even necessary so **it would be really useful to add at least VPS together with Azure to the docs as an alternative (given that it's an option)**! so people who just want to get started with the tech would know whether it's even feasible to run Blazor Server App without Azure and maybe provide some numbers like average concurrent connections, server load and such on multiple setups to give people a sense of the system requirement, I'm just thinking out load here.

ATM because I can't find any data I'm thinking of using Rust or just Vue (with JS) which would be a shame because I like the .NET ecosystem and I've been using it for many years and this is an opportunity to start my first Blazor app which I was really excited about until I read about SignalR, concurrency and Azure with zero numbers to tell me anything useful so I can make an informed decision and decide whether I want to go forward with it or not. 😃 

I've posted a similar thing over Discord/DotNetEvolution/Blazor (not sure if it's an official channel but based on the people there it seems like it is) and got no answer to my questions. 

Any advice or guidance on the matter would be appreciated.

**Comment by iam3yal** [comment, marked as answer]
Never mind, I've found this [link](https://docs.microsoft.com/en-us/aspnet/core/signalr/scale?view=aspnetcore-5.0) which answered some of my questions.

I've made a free Azure account and hopefully Microsoft have a plan that fits my needs and is not too costly to start with.

**Reply by Pilchie** [reply]
There is nothing that ties Blazor (including Blazor Server) to Azure.  It's certainly *easy* to use the Azure SignalR Service for the SignalR portion, but you can definitely host the SignalR part on your own server.  This is what happens when you run locally for example.

**Reply by iam3yal** [reply]
Thank you for the reply.
